### PR TITLE
Portfolio-10: Preserve data in the browser

### PR DESCRIPTION
**Pull request by danifromecuador**
## Portfolio-10: Preserve data in the browser using local storage
In this pull request I have:
- Add new classes to name, email and textarea fields in HTML
- Create an objet to storage the user's data
- Verify if there is previous data
- Save the user's data using local storage
- Fill the name, email and textarea of the user with their previous data

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -206,9 +206,9 @@ <h2 class="primary-text">About me</h2>
...
       <form class="form" method="post" action="https://formspree.io/f/xnqrekyo">
         <label for="name"></label>
-        <input type="text" class="single input-field" id="name" name="name" maxlength="30" placeholder="Full name" required>
+        <input type="text" class="single input-field userData" id="name" name="name" maxlength="30" placeholder="Full name" required>
         <label for="email"></label>
-        <input type="email" class="single input-field" id="email" name="email" placeholder="Email address" required>
+        <input type="email" class="single input-field userData" id="email" name="email" placeholder="Email address" required>
         <label for="message"></label>
         <textarea class="input-field" id="message" name="message" rows="3" maxlength="500" placeholder="Enter text here" required></textarea>
         <input class="see-project" id="submit" type="submit" value="Get in touch">
```
Comment (jerryowusu) [inline index.html]: Nice implementation so far. :+1: However, the data for all the three inputs including the textarea should be populated and stored in the local storage, but I see the implementation on only the name and email inputs. Kindly add the textarea input. :wink:

File: script.js [hunk 2]
```diff
@@ -151,3 +151,23 @@ submit.addEventListener('click', (event) => {
...
+  });
+}
+
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  userInput.forEach((input) => {
+    data[input.name] = input.value;
+  });
+  localStorage.setItem('userData', JSON.stringify(data));
+});
```
Comment (jerryowusu) [inline script.js]: Great effort so far. However, I can see a few things you need to update in order to match the project requirements. 

- [x] Currently, the data is not saved to the local storage unless you click on the `Get in Touch` button hence the submit button does not work even if the inputs are correct which is not the best way. Kindly ensure that the submit button works correctly and the data is saved into the local storage when they are typed in the input fields without clicking any button. 
- [x] If at the previous visit to the page, in the same browser, no input was entered in the form then the form inputs should remain empty. However, I see that the data in the localstorage persists and it is removed only when you clear them using the clear all button in the browser console. 

Kindly check and fix this issues to make your project one of the best and up to standard. :pray:

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @danifromecuador & @Ruthmy , 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

almost-there-reach


## Highlights
Great effort so far :heavy_check_mark: 
No linter errors :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
# STATUS: APPROVED ♻️♻️
Hi @danifromecuador 

To highlight 👍🏽
✔️ Navigations work as expected
✔️ No linter errors

Your project is complete! There is nothing else to say other than... it's time to merge it 🚀
Congratulations! 🎉

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**